staging/src/k8s.io/apiserver/pkg/server/filters/with_early_late_annotations.go

```diff
@@ -64,16 +65,22 @@ func exemptIfHealthProbe(r *http.Request) bool {
 	return false
 }
 
-// WithShutdownLateAnnotation, if added to the handler chain, tracks the
-// incoming request(s) after the apiserver has initiated the graceful
-// shutdown, and annoates the audit event for these request(s) with
-// diagnostic information.
-// This enables us to identify the actor(s)/load balancer(s) that are sending
-// requests to the apiserver late during the server termination.
-// It should be placed after (in order of execution) the
-// 'WithAuthentication' filter.
-func WithShutdownLateAnnotation(handler http.Handler, shutdownInitiated lifecycleEvent, delayDuration time.Duration) http.Handler {
-	return withShutdownLateAnnotation(handler, shutdownInitiated, delayDuration, exemptIfHealthProbe, clockutils.RealClock{})
+// WithShutdownResponseHeader, if added to the handler chain, adds a header
+// 'X-OpenShift-Disruption' to the response with the following information:
+//
+//	shutdown={true|false} shutdown-delay-duration=%s elapsed=%s host=%s
+//	 shutdown: whether the server is shutting down gracefully.
+//	 shutdown-delay-duration: value of --shutdown-delay-duration server run option
+//	 elapsed: how much time elapsed since the server received the TERM signal
+//	 host: host name of the server, it is used to identify the server instance
+//	       from the others.
+//
```

Comment:
Might be nice to have a Go API for reading and writing these header values somewhere in apiserver-library-go, depending on how widely this is used. How likely is it that we add more fields to this, especially ones potentially containing whitespace or arbitrary strings?

Reply:
sure, maybe when we extend this to be included as a round tripper inside kcm/scheduler/kubelet or any other operator.